src/bootchooser.c

```diff
@@ -333,26 +387,33 @@ static gboolean uboot_set_state(RaucSlot *slot, gboolean good) {
 
 /* Set slot as primary boot slot */
 static gboolean uboot_set_primary(RaucSlot *slot) {
-	GString *order = g_string_sized_new(10);
+	GString *order_new = g_string_sized_new(10);
+	GString *order_current = NULL;
+	gchar **bootnames = NULL;
 	gboolean res = FALSE;
 	gchar *key = NULL;
-	GList *slots;
 
 	g_assert_nonnull(slot);
 
-	g_string_append(order, slot->bootname);
+	g_string_append(order_new, slot->bootname);
 
-	/* Iterate over class members */
-	slots = g_hash_table_get_values(r_context()->config->slots);
-	for (GList *l = slots; l != NULL; l = l->next) {
-		RaucSlot *s = l->data;
-		if (s == slot)
+	res = uboot_env_get("BOOT_ORDER", &order_current);
```

Comment:
Currently, we don't require that BOOT_ORDER already exists in the env. Just use the list of bootable slots as a default for order_current.